classes/SpecificPriceRule.php

```diff
@@ -275,8 +275,11 @@ public function getAffectedProducts($products = false)
                     $query->select('NULL as `id_product_attribute`');
                 }
 
+                // Merge previous result to current results
                 $result = array_merge($result, Db::getInstance()->executeS($query));
             }
+            // Remove duplicate after the array_merge
+            $result = array_intersect_key($result, array_unique(array_map('serialize', $result)));
```

Comment:
```suggestion
            $result = array_unique($result, SORT_REGULAR);
```

It should be enough, if you can try